Platform 553

awalsh272 opened the pull request:
Being able to aggregate on arbitrary columns

awalsh272 committed:
Merge pull request #8 from jamccomb92/dev

Dev

awalsh272 committed:
test comment

awalsh272 committed:
test comment 2

hopetambala committed:
Merge pull request #12 from jamccomb92/dev

Dev

hopetambala committed:
Update README.md

awalsh272 committed:
Merge pull request #6 from jamccomb92/PLATFORM-486

Platform 486

awalsh272 committed:
Merge pull request #16 from awalsh272/andy-dev

Andy dev

awalsh272 committed:
Revert "Andy dev"

awalsh272 committed:
Merge pull request #17 from puente-dr/revert-16-andy-dev

Revert "Andy dev"

hopetambala committed:
Revert "Merge pull request #17 from puente-dr/revert-16-andy-dev"

This reverts commit f440fdee84bc50c00017f2ee2e33ca1ccff9a597, reversing
changes made to a0d57b91bc519e3f07fd0bc798b4a27beb38c173.

bcranteverfi committed:
Test deploy.

bcranteverfi committed:
Test deploy.

bcranteverfi committed:
Test deploy.

bcrant committed:
Run Black code formatter.

bcrant committed:
Upgrade to Python 3.9.5 and update dependencies.

bcrant committed:
Reformat code, add TODOs, quick refactors.

bcrant committed:
Merge pull request #1 from bcrant/chore/update_deps

Chore/update deps

bcrant committed:
chore/update_deps. Update cloudformation deploy template with python3.9

bcrant committed:
Apply Black formatter. Refactor Index.py.

bcrant committed:
Apply Black formatter and refactor lib scripts.

bcrant committed:
Merge branch 'dev' into chore/rebase-dev

bcrant committed:
gitignore .DS_Store

bcrant committed:
Checkpoint.

bcrant committed:
Begin denormalizing Survey Forms into Questions and Answers with IDs.

bcrant committed:
Join denormalized Questions and Answers.

bcrant committed:
Readability refactors to restCall. Place customFormSchema within ecosystem, rename functions, refactor comments.

bcrant committed:
Checkpoint. Batch export data using direct connection to Back4App's MongoDB.

bcrant committed:
Write MongoDB tables to S3 as JSON.

bcrant committed:
Write MongoDB tables to S3 as Python Dictionaries.

bcrant committed:
Move project to own puente-etl dir. Refactor components into ETL paradigm.

bcrant committed:
Refactor as modular components. Add pickled list as S3 output serialization type.

bcrant committed:
Refactor transforms to operate on all data instead of a given subset (objectId).

bcrant committed:
Finish Form Specifications batch transformation. Begin transforming Form Results. Remove tabulate. Move to factory pattern for execution. Still using PyCharm run configurations for dev execution in the meantime.

bcrant committed:
Repurpose some Form Specifications transformation steps for the Form Results.

bcrant committed:
Ignore csv files in repo. Remove mostly null and noisy columns from outputs.

bcrant committed:
Remove dev logging.

bcrant committed:
Aggregate and output denormalized Form Results.

hopetambala committed:
Merge branch 'master' into dev

bcrant committed:
PR Review

bcrant committed:
Remove tables from extract. Add example .env.

hopetambala committed:
Merge branch 'PLATFORM-539-CustomForms' of https://github.com/bcrant/serverless into bcrant-PLATFORM-539-CustomForms

awalsh272 committed:
feat: pass a list to aggregate by

Pass a list of columns to get_form_results function to aggregate custom form answers by. Needed to split form_result_p_client and merge _id from survey data on part of the split column

awalsh272 committed:
chore: remove print statements

hopetambala commented on puente-etl/transform_form_results.py:
```diff
@@ -0,0 +1,173 @@
+
+    df[["Survey_col", "merge_id"]] = df["form_result_p_client"].str.split("$", expand=True)
+
+    # df.to_csv('./denormalized_form_results.csv', index=False)
```
@joseph-mccombs don't we just have to send this to the right nested form bucket in s3?

hopetambala commented on puente-etl/transform_form_results.py:
```diff
@@ -0,0 +1,173 @@
+        )['answer_value'] \
+        .value_counts() \
+        .rename(columns={'count': 'answer_count'})
+    # agg_df.to_csv('./aggregated_results.csv', index=False)
```
All we need to do is push these results to one bucket and create a link

awalsh272 committed:
chore: making jobs.py work

Fix a couple of silly errors to get it to run

awalsh272 committed:
feat: fixed (some?) of the form merging issues

The previous aggregation setup caused a significant drop in the number of forms being returned. Removing custom_form_active from the groupby and removing rows with custom_form_active='false' seems to have done the trick, though its worth checking the other columns as well
